x-pack/plugins/security_solution/public/overview/containers/overview_risky_host_links/use_risky_hosts_dashboard_button_href.ts

```diff
@@ -16,13 +16,15 @@ export const DASHBOARD_REQUEST_BODY = {
 };
 
 export const useRiskyHostsDashboardButtonHref = (to: string, from: string) => {
-  const createDashboardUrl = useKibana().services.dashboard?.dashboardUrlGenerator?.createUrl;
-  const savedObjectsClient = useKibana().services.savedObjects.client;
+  const {
+    dashboard,
```

Comment:
Can you further destructure locator out of here, since I think that's the only part of dashboard that it uses.  Then the hook can only have a dependency on the locator?

Reply:
Unfortunately we can't destructure locator out of dashboard, because the dashboard service is optional.